api/query/atoms_test.go

```diff
@@ -185,18 +185,343 @@ func TestStructuredQuery_nested(t *testing.T) {
 		}
 	}`), &rq)
 	assert.Nil(t, err)
-	assert.Equal(t, runQuery{
+	assert.Equal(t, RunQuery{
 		runIDs: []int64{0, 1, 2},
-		query: or{
-			or: []query{
+		AbstractQuery: or{
+			or: []AbstractQuery{
 				and{
-					and: []query{
-						not{testNamePattern{"cssom"}},
-						testNamePattern{"html"},
+					and: []AbstractQuery{
+						not{TestNamePattern{"cssom"}},
+						TestNamePattern{"html"},
 					},
 				},
 				testStatusConstraint{"edge", shared.TestStatusValueFromString("TIMEOUT")},
 			},
 		},
 	}, rq)
 }
+
+func TestStructuredQuery_bindPattern(t *testing.T) {
+	tnp := TestNamePattern{
+		Pattern: "/",
+	}
+	q := tnp.BindToRuns([]shared.TestRun{})
+	assert.Equal(t, tnp, q)
+}
+
+func TestStructuredQuery_bindStatusNoRuns(t *testing.T) {
+	assert.Equal(t, True{}, testStatusConstraint{
+		browserName: "Chrome",
+		status:      1,
+	}.BindToRuns([]shared.TestRun{}))
+}
+
+func TestStructuredQuery_bindStatusSingleRun(t *testing.T) {
+	assert.Equal(t, RunTestStatusConstraint{
```

Comment:
nit: assign `expected` and `actual` params to locally well-named variables so the reader can tell what should be equal.
I think (?) that `firstRun` and `firefox` make sense. But I'm not sure.. hence the request ;)